packages/wallet-frontend/src/lib/hooks/useDebounce.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useRef, useState } from 'react'
+
+function useDebounce<T>(value: T, delay: number) {
+  const [debouncedValue, setDebouncedValue] = useState(value)
+
+  const lastValue = useRef<T>(value)
+  const [isLoading, setIsLoading] = useState(false)
+
+  useEffect(() => {
+    if (value !== '' && lastValue !== value) {
+      setIsLoading(true)
+      const handler = setTimeout(() => {
+        setIsLoading(false)
+        setDebouncedValue(value)
+      }, delay)
+
+      return () => {
+        setIsLoading(false)
```

Comment:
Is this needed? Wouldn't `setIsLoading` that is placed inside `handler` cover everything?